src/window/building/depot.c

```diff
@@ -0,0 +1,370 @@
+#include "depot.h"
+
+#include "building/building.h"
+#include "building/storage.h"
+#include "building/warehouse.h"
+#include "figure/figure.h"
+#include "graphics/generic_button.h"
+#include "graphics/image.h"
+#include "graphics/lang_text.h"
+#include "graphics/panel.h"
+#include "graphics/scrollbar.h"
+#include "graphics/text.h"
+#include "graphics/window.h"
+#include "translation/translation.h"
+#include "window/building_info.h"
+
+static void order_set_source(int index, int param2);
+static void order_set_destination(int index, int param2);
+static void order_set_resource(int index, int param2);
+static void order_set_condition_type(int index, int param2);
+static void order_set_condition_threshold(int index, int param2);
+static void set_order_source(int depot_building_id, int building_id);
+static void set_order_destination(int depot_building_id, int building_id);
+static void set_order_resource(int depot_building_id, int resource_id);
+
+static struct {
+    int focus_button_id;
+    int orders_focus_button_id;
+    int resource_focus_button_id;
+    int storage_building_focus_button_id;
+    int depot_resource_focus_button_id;
+} data;
+
+static const translation_key TRANSLATION_KEY_ORDER_CONDITION[4] = {
+    TR_ORDER_CONDITION_NEVER,
+    TR_ORDER_CONDITION_ALWAYS,
+    TR_ORDER_CONDITION_SOURCE_HAS_MORE_THAN,
+    TR_ORDER_CONDITION_DESTINATION_HAS_LESS_THAN
+};
+
+#define DEPOT_BUTTONS_X_OFFSET 32
+#define DEPOT_BUTTONS_Y_OFFSET 204
+static generic_button depot_order_buttons[] = {
+    {100, 0, 26, 26, order_set_resource, button_none, 1, 0},
+    {100, 56, 284, 22, order_set_source, button_none, 2, 0},
+    {100, 82, 284, 22, order_set_destination, button_none, 3, 0},
+    {100, 30, 284, 22, order_set_condition_type, button_none, 4, 0},
+    {384, 30, 32, 22, order_set_condition_threshold, button_none, 5, 0},
+};
+
+static int storage_buildings_count() {
```

Comment:
On functions, the `{` is on a different line. Also, don't forget the `void` on the paramenter list.